### PR TITLE
[8.2] Group share etag propagation 

### PR DESCRIPTION
**Pull request by icewind1991**
For #20714

~~Currently only unit test for it~~

### CODE REVIEW

File: apps/files_sharing/tests/etagpropagation.php [hunk 1]
```diff
@@ -411,7 +341,7 @@ public function testRecipientUploadInDirectReshare() {
 		$this->assertEtagsNotChanged([self::TEST_FILES_SHARING_API_USER3]);
 		$this->assertEtagsChanged([self::TEST_FILES_SHARING_API_USER1, self::TEST_FILES_SHARING_API_USER2, self::TEST_FILES_SHARING_API_USER4]);
 
-		$this->assertAllUnchaged();
+		$this->assertAllUnchanged();
```
Comment (PVince81) [inline apps/files_sharing/tests/etagpropagation.php]: looks like this will have to be forward ported to master too

**Comment by PVince81** [comment]
Thanks a lot!

I guess the new unit tests should later be ported to master as well for additional benefit.

**Comment by icewind1991** [comment]
I'm having problems reproducing the propagation problem from the original issue

**Comment by icewind1991** [comment]
Ok, I managed to reproduce a failure by re-sharing a subfolder with a group

**Comment by icewind1991** [comment]
Ok, fixed the problem, ready for review

**Comment by icewind1991** [comment]
https://github.com/owncloud/core/pull/21044/files#diff-40d7c423bcba1cd2aec6e34c1c9c74bcR157 is the main fix

**Comment by PVince81** [comment]
Smashbox still failing on CI for some reason ?

**Comment by icewind1991** [comment]
One less failure at least

**Comment by PVince81** [comment]
I cherry picked locally https://github.com/owncloud/core/pull/20949 (backport pending) to make sure the test actually pass.
lib/owncloud/test_sharePropagationInsideGroups.py does pass.

:+1:

**Comment by PVince81** [comment]
@karlitschek please approve this non-backport for 8.2.2. This is a 8.2 specific fix for etag propagation for groups. The 9.0 fix is different and cannot be backported as is.

**Comment by PVince81** [comment]
Needs a second reviewer @MorrisJobke @jvillafanez @DeepDiver1975 @nickvergessen @rullzer @schiesbn

**Comment by karlitschek** [comment]
please backport :+1:

**Comment by rullzer** [comment]
awesome! Looks good and seems to fix the issue. :+1:

File: apps/files_sharing/tests/etagpropagation.php (hunk 1 above)
Comment (nickvergessen) [inline apps/files_sharing/tests/etagpropagation.php]: fixing a method name... of tests... in a maintenance branch ....

**Comment by PVince81** [comment]
@icewind1991 can you forward port the unit test ?

**Pull request merged**